### PR TITLE
Add Prefix "n/" for specifying names #467

### LINKED ISSUE #467: Add Prefix "n/" for specifying names
Putting the name in the preamble can make the code more complex. For example, in EditCommandParser, the preamble includes both the index and the name, so EditCommandParser has to split it.

Let's add a new Prefix "n/" for specifying names.

### PR DESCRIPTION
**Pull request by limmlingg**
Fixes #467 
Part of #407

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@limmlingg submitted v1 for review.

* [v1 1/3] [Add Prefix "n/" for specifying names](https://github.com/se-edu/addressbook-level4/pull/474/commits/616fb9dd0511ed5ed67b4789b69ddd7f39ac4b7b)
* [v1 2/3] [Remove ParserUtil\#split\(String\, int\)](https://github.com/se-edu/addressbook-level4/pull/474/commits/3d8c5fc9c73e8ddaa233c436898a0f9e46c339f9)
* [v1 3/3] [Add new information about Name Prefix to docs](https://github.com/se-edu/addressbook-level4/pull/474/commits/adc722fde55f1d86cf356421da67f463dd6cc954)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/dfe3b25eb274119f2d0605fa266877597a3341c6...adc722fde55f1d86cf356421da67f463dd6cc954))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/474/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 1]
```diff
@@ -59,16 +60,11 @@
     private EditCommandParser parser = new EditCommandParser();
 
     @Test
-    public void parse_noValidPrefix_everythingTakenAsName() throws Exception {
-        // Unable to test whether complex strings such as "1 q/unknown is this k/parsed"
-        // are entirely parsed as Name. Verification of Name field requires successful parsing.
-        // However, "/" is an invalid character for Name, thus parsing invalid prefixes will always fail.
-        // E.g, we cannot verify if "q/unknown" is parsed as Name, resulting in an error, or whether
-        // "q/unknown is this k/parsed" is parsed as Name, which results in an error as well.
+    public void parse_noValidPrefix_failure() throws Exception {
```
Comment (Zhiyuan-Amos) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: i think this test is unnecessary

Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: merge with invalid index. "1 some random i/ string"

**Comment by CanIHasReview** [comment]
# v2

@limmlingg submitted v2 for review.

* [v2 1/3] [Add Prefix "n/" for specifying names](https://github.com/se-edu/addressbook-level4/pull/474/commits/5a33659710c3efd45ca14136559321045f4e7261)
* [v2 2/3] [Remove ParserUtil\#split\(String\, int\)](https://github.com/se-edu/addressbook-level4/pull/474/commits/f9bafa07e40317b876437e1fae4642c4b0164be2)
* [v2 3/3] [docs: Add new Name prefix](https://github.com/se-edu/addressbook-level4/pull/474/commits/821784d1843ccbd5609ecd88cd131ecfd08ad988)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/3de3b34bcf05aa4a9a656b238be22d7de6d78279...821784d1843ccbd5609ecd88cd131ecfd08ad988)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/41c22b700a5302acbd620fda176d0292f354c272~1...41c22b700a5302acbd620fda176d0292f354c272))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/474/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v3

@limmlingg submitted v3 for review.

* [v3 1/3] [Add Prefix "n/" for specifying names](https://github.com/se-edu/addressbook-level4/pull/474/commits/31202dc7c1977e0b813db0ef1e2c6bbca1c5494b)
* [v3 2/3] [Remove ParserUtil\#split\(String\, int\)](https://github.com/se-edu/addressbook-level4/pull/474/commits/7b939245dce021806826d23c2e0192426572e960)
* [v3 3/3] [docs: Add new Name prefix](https://github.com/se-edu/addressbook-level4/pull/474/commits/ac4f42a7e6b4d69aa20549a0b07cbdd56cd7e0a8)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/3de3b34bcf05aa4a9a656b238be22d7de6d78279...ac4f42a7e6b4d69aa20549a0b07cbdd56cd7e0a8)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/8890d0a91435bfda538ce80aa91f1a403b9c2a3f~1...8890d0a91435bfda538ce80aa91f1a403b9c2a3f))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/474/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by limmlingg** [comment]
@damithc The User Guide states that `Parameters can be in any order.`

**Comment by yamgent** [comment]
@limmlingg This will need a rebase.

**Comment by CanIHasReview** [comment]
# v4

@limmlingg submitted v4 for review.

* [v4 1/3] [Add Prefix "n/" for specifying names](https://github.com/se-edu/addressbook-level4/pull/474/commits/17aceb2eb3fb492aa8541c6224d530a2836b3777)
* [v4 2/3] [Remove ParserUtil\#split\(String\, int\)](https://github.com/se-edu/addressbook-level4/pull/474/commits/3c729eccff474ae0f36636207807a26df52eb3f2)
* [v4 3/3] [docs: Add new Name prefix](https://github.com/se-edu/addressbook-level4/pull/474/commits/8c3eb78925eff1fb77e38cd988b9b1e1a14aac57)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/8e4f02b5ad8da2c440f262c2787f24f824978e4c...8c3eb78925eff1fb77e38cd988b9b1e1a14aac57)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/8deaf13939502175c413da4c437856cb5bd13920~1...8deaf13939502175c413da4c437856cb5bd13920))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/474/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by yamgent** [comment]
@damithc Need your approval for this PR.

**Comment by yamgent** [comment]
@limmlingg Can you propose a merge commit message?

**Pull request merged**